File: app/src/main/java/com/bq/zowi/models/commands/LedCommand.java
package com.bq.zowi.models.commands;

import com.bq.zowi.models.commands.Command;

/* JADX INFO: loaded from: classes.dex */
public class LedCommand extends MotionlessCommand {
    private static final String LED_MOUTH = "L 00";
    private static final Command.Action[] LED_MOUTH_ACTIONS = {Command.Action.LED_MOUTH};
    private String ledMatrix;

    public LedCommand(String ledMatrix) {
        setAction(Command.Action.LED_MOUTH);
        this.ledMatrix = ledMatrix;
    }

    @Override // com.bq.zowi.models.commands.Command
    public Command.Action[] getAllowedActions() {
        return LED_MOUTH_ACTIONS;
    }

    @Override // com.bq.zowi.models.commands.Command
    public String getCommandValue() {
        return LED_MOUTH + this.ledMatrix + Command.CRLN;
    }

    public String getLedMatrix() {
        return this.ledMatrix;
    }

    @Override // com.bq.zowi.models.commands.Command
    public Command copy() {
        return new LedCommand(this.ledMatrix);
    }
}

File: zowi-core/src/main/kotlin/com/bq/zowi/models/commands/LedCommand.kt
package com.bq.zowi.models.commands

class LedCommand(private val ledMatrix: String) : MotionlessCommand() {

    init {
        setAction(Command.Action.LED_MOUTH)
    }

    override fun getAllowedActions(): Array<Command.Action> = LED_MOUTH_ACTIONS
    override fun getCommandValue(): String = "${LED_MOUTH}${ledMatrix}${Command.CRLN}"
    fun getLedMatrix(): String = ledMatrix
    override fun copy(): Command = LedCommand(ledMatrix)

    companion object {
        private const val LED_MOUTH = "L 00"
        private val LED_MOUTH_ACTIONS = arrayOf(Command.Action.LED_MOUTH)
    }
}
